### PR TITLE
Feature/JS-4872: New messages divider

### PR DESCRIPTION
**Pull request by mhlv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/ts/component/block/chat.tsx [hunk 1]
```diff
@@ -136,10 +142,22 @@ const BlockChat = observer(class BlockChat extends React.Component<I.BlockCompon
...
+
+		if (lastMessageId && this.messagesMap[lastMessageId]) {
+			const node = this.messagesMap[lastMessageId].node;
+
+			this.lastMessageId = lastMessageId;
+			this.lastMessageOffset = node.offsetTop;
+			this.scrollToMessage(lastMessageId);
+		};
+
+		this.getScrollContainer().on('scroll.chat', e => this.onScroll(e));
```
Comment (ra3orblade) [inline src/ts/component/block/chat.tsx:159]: Better use scroll.[blockId] + U.Common.getEventNamespace(isPopup);

File: src/ts/component/block/chat.tsx [hunk 2]
```diff
@@ -407,6 +426,10 @@ const BlockChat = observer(class BlockChat extends React.Component<I.BlockCompon
 		return this.state.threadId || this.props.block.id;
 	};
 
+	getMessageNodeById (id) {
```
Comment (ra3orblade) [inline src/ts/component/block/chat.tsx:429]: Remove

File: src/ts/component/block/chat.tsx [hunk 3]
```diff
@@ -489,16 +514,62 @@ const BlockChat = observer(class BlockChat extends React.Component<I.BlockCompon
...
+		const form = $('#formWrapper');
+		const formPadding = Number(form.css('padding-bottom').replace('px', ''));
+		const viewport = container.outerHeight() - form.height() - formPadding;
+		const messages = this.getMessages();
+
+		const messagesIntoView = messages.filter((it) => {
+			const node = this.messagesMap[it.id].node;
+			const oT = node.offsetTop + node.clientHeight;
+
+			return oT >= top && oT < top + viewport;
```
Comment (ra3orblade) [inline src/ts/component/block/chat.tsx:530]: Add parenthesis to logical parts

File: src/ts/component/block/chat.tsx [hunk 4]
```diff
@@ -489,16 +514,62 @@ const BlockChat = observer(class BlockChat extends React.Component<I.BlockCompon
...
+
+		if (lastNode.offsetTop > this.lastMessageOffset) {
+			this.lastMessageId = last.id;
+			this.lastMessageOffset = lastNode.offsetTop;
+			Storage.setLastChatMessageId(blockId, last.id);
+		};
+	};
+
+	scrollToMessage (id: string) {
+		window.setTimeout(() => {
```
Comment (ra3orblade) [inline src/ts/component/block/chat.tsx:548]: Why timeout?

File: src/ts/component/block/chat.tsx [hunk 5]
```diff
@@ -489,16 +514,62 @@ const BlockChat = observer(class BlockChat extends React.Component<I.BlockCompon
...
 			const height = isPopup ? container.get(0).scrollHeight : document.body.scrollHeight;
 
 			container.scrollTop(height);
 		}, 10);
 	};
 
+	getScrollContainer () {
+		const { isPopup } = this.props;
+
+		return U.Common.getScrollContainer(isPopup);
```
Comment (ra3orblade) [inline src/ts/component/block/chat.tsx:570]: This is redundant, no need for such method

**Review by ra3orblade — CHANGES_REQUESTED** [review]

File: src/ts/component/block/chat.tsx (hunk 4 above)
Comment (mhlv) [reply src/ts/component/block/chat.tsx:548]: delay to let messages render correctly. moved it to the call

**Review by ra3orblade — APPROVED** [review]

**Pull request merged**